### ISSUE TITLE
adding a bluilt in pitch analyzer to the sample track

### ISSUE THREAD

**Issue by Reaper10** [LMMS/lmms#1816]
I think it would be cool for the sample track to have a built in pitch analyzer for tuning of live  Instruments like Guitars. The reason  for it to be built in and not as a plugin is I think it's silly to have it on all of the tracks when you only need it on the sample track. 

How to:
Go to the gear menu click on show/ hide the tunner.

Tunner GUI
![tuner](https://cloud.githubusercontent.com/assets/8137561/6469410/ee5cf098-c1a1-11e4-87b4-62e9387aa8c9.png)
#1471   This will enhance the sample track even more. The pitch analyzer should be built into the track not as a plugin this way all sample tracks have it and you don't need waist a plugin spot. When the LMMS pitch analyzer plugin is built It all so can be built in to the audio track editor with a on off button. Both the built in pitch analyzer and the plugin versions should be in LMMS for three reasons.

Reasons this is good:
1. Backward compatibility.
2. When a user is using the audio track editor for recording a user doesn't need anything but the audio track editor window to check the tuning.
3. The plugin version would be great for those who want to use LMMS live and want to check the tuning without needing to open the audio track editor window.

**Comment by Sti2nd** [comment]
I fail to see what this has to do with the sample track, at least when considering how regular tuners work.

**Comment by badosu** [comment]
@Sti2nd I guess what he wants is not exactly a tuner but a pitch analyzer.

**Comment by Sti2nd** [comment]
Ok. Probably a VST or LADSPA for that but then they need to access the sample track indeed.

**Comment by Reaper10** [comment]
The pitch analyzer should be built into the track not as a plugin.

**Comment by Sti2nd** [comment]
OK, sure, nice idea

**Comment by tresf** [comment]
> #1471 This will enhance the sample track even more. The pitch analyzer should be built into the track not as a plugin this way all sample tracks have it and you don't need waist a plugin spot.

This is an interesting request... I've utilized sample tracks many times and never had a need for this, but I also don't do much with live instruments.  LMMS still lacks quite a bit as a sample-track mixer, so this just goes into a laundry list of enhancements.

However, I don't know if your use-case warrants breaking our defaults.

Instead, we should support the idea of a "Sample Track Preset" (which should be pretty easy to implement actually).  It will still take a slot on the FX chain though.

Can you agree that a preset could address this?

**Comment by badosu** [comment]
Aren't there already any VSTs that do pitch analysis? You could just plug one of them into the effects channel/audiofileprocessor.

Also, if there are not free Pitch Analysis VSTs on the market, creating a lv2 plugin for this would be the best way to do it (assuming we get lv2 support into LMMS).

**Comment by tresf** [comment]
> Aren't there already any VSTs that do pitch analysis? You could just plug one of them into the effects channel/audiofileprocessor.

Yeah, IIRC that's what he's doing now.

> if there are not free Pitch Analysis VSTs on the market, creating a lv2 plugin for this would be the best way to do it (assuming we get lv2 support into LMMS).

:)

**Comment by Spekular** [comment]
I think it's silly to make users have a pitch analyzer on sample tracks if
they don't want to, and not allow them to have it on other instruments. If
we add this I vote we do it as a plugin.

**Comment by Spekular** [comment]
Speaking of effects and sample tracks, is it possible to add sample tra

**Comment by Spekular** [comment]
-_- Was gonna cancel that message and ask it somewhere else but oh well. Is
it possible to add sample tracks to fx channels?

**Comment by curlymorphic** [comment]
> Is it possible to add sample tracks to fx channels?

no, but they do have there own effects chain, double clicking on the sample track label in the song editor opens this.

**Comment by Spekular** [comment]
@curlymorphic I'm aware that you can add effects to sample tracks, but
thanks anyways :)

**Comment by Reaper10** [comment]
There is no need for a LMMS pitch analyzer plugin with the Free Music Instrument Tuner witch is soon going to be a LV2 plugin and the it's stand alone software for windows, linix and coming soon osx.

**Comment by Reaper10** [comment]
https://github.com/x42/tuna.lv2
https://github.com/gillesdegottex/fmit
Here some other tuner options----------^
